cvpy/biomedimage/BiomedImage.py

```diff
@@ -78,45 +78,167 @@ def quantify_sphericity(self, image_table: ImageTable, use_spacing: bool, input_
+                             dict(name=mask.resolution, rename="res"),
+                             dict(name=mask.imageFormat, rename="form")]
+
+            # SQL string to create the mask table
+            fed_sql_str = f'''create table images_to_mask {{options replace=true}} as 
+                select a.seg, a.dim, a.res, a.form, b.* 
+                from {mask.table.name} as a inner join {image.table.name} as b 
+                on a._id_=b._id_'''
+
+            # Dictionary for specifying information in our binary operation
+            binary_operation_dict = dict(binaryoperationtype="mask_specific",
+                                         image="seg",
+                                         dimension="dim",
+                                         resolution="res",
+                                         imageformat="form",
+                                         outputBackground=output_background,
+                                         inputBackground=input_background)
+
+            # Mini method for renaming columns in ImageTable object
+            # *This step must be completed AFTER the alter table command*
+            def rename_columns():
+                mask.image = "seg"
+                mask.dimension = "dim"
+                mask.resolution = "res"
+                mask.imageFormat = "form"
+
+        ###############################################
+        ############# Mask Tbl Encoded ################
+        ###############################################
+        else:
+
+            # List of columns that will be alternated
+            alter_columns = [dict(name=mask.image, rename="seg")]
+
+            # SQL string to create the mask table
+            fed_sql_str = f'''create table images_to_mask {{options replace=true}} as 
+                select a.seg, b.* 
+                from {mask.table.name} as a inner join {image.table.name} as b 
+                on a._id_=b._id_'''
+
+            # Dictionary for specifying information in our binary operation
+            binary_operation_dict = dict(binaryoperationtype="mask_specific",
+                                         image="seg",
+                                         outputBackground=output_background,
+                                         inputBackground=input_background)
+
+            # Mini method for renaming columns in ImageTable Object
+            # *This step must be completed AFTER the alter table command*
+            def rename_columns():
+                mask.image = "seg"
+
+        ###############################################
+        ############### Masking Step ##################
+        ###############################################
+
+        # Change column names so that we can merge tables
+        conn.table.altertable(name=mask.table, columns=alter_columns)
+
+        # Create Images to Mask Table
+        images_to_mask = conn.CASTable("images_to_mask", replace=True)
```

Comment:
Is there any concern that choosing a table name like this could *potentially* overlap with a customers table? Should we use underscores around it?

Reply:
That's a great point! I added underscores to this table name.